### PR TITLE
Allow using docker secrets for env variables

### LINKED ISSUE #52: Allow using Docker secrets for Environment Variables
* [x] I've read the [guidelines for Contributing to YOURLS Projects](https://github.com/YOURLS/.github/blob/master/CONTRIBUTING.md)
* [x] This request isn't a duplicate of an existing request
* [x] This is not a personal support request that should be posted on the [YOURLS Discourse](https://discourse.yourls.org/) community

**Summary**
Many docker containers allow using Docker Secrets to pass in environment variables. It would be great to be able to input configuration through secrets.

**Motivation**
The mysql container can use secrets for its configuration. Using the same method of configuration for both would be great.

**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Be able to append "_FILE" to any of the environment variables to be able to use the /var/secrets files as the source of the values.

### PR DESCRIPTION
**Pull request by jhollowe**
copied in file_env() from the mysql entrypoint and added appropriate calls

not setting defaults, but could modify to set some defaults for the env vars.

closes #52

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+	local var="$1"
+	local fileVar="${var}_FILE"
+	local def="${2:-}"
+	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+		mysql_error "Both $var and $fileVar are set (but are exclusive)"
```
Comment (LeoColomb) [inline docker-entrypoint.sh:13]: `mysql_error` function is not defined. Maybe just an echo?

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+	export "$var"="$val"
+	unset "$fileVar"
+}
+
+file_env 'YOURLS_DB_HOST'
+file_env 'YOURLS_DB_USER'
+file_env 'YOURLS_DB_PASS'
+file_env 'YOURLS_DB_NAME'
+file_env 'YOURLS_DB_PREFIX'
+file_env 'YOURLS_COOKIEKEY'
```
Comment (LeoColomb) [inline docker-entrypoint.sh:30]: `YOURLS_COOKIEKEY` should be excluded.

File: docker-entrypoint.sh (hunk 1 above)
Comment (jhollowe) [reply docker-entrypoint.sh:13]: copy/paste will get you every time 😄

Copied from https://github.com/docker-library/mysql/blob/master/8.0/docker-entrypoint.sh#L16
do we want all the information at the referenced https://github.com/docker-library/mysql/blob/master/8.0/docker-entrypoint.sh#L8 ?

File: docker-entrypoint.sh (hunk 2 above)
Comment (jhollowe) [reply docker-entrypoint.sh:30]: If it shouldn't be here, should It be removed from the readme as well? I just pulled all the env vars from this container's readme.

File: docker-entrypoint.sh (hunk 1 above)
Comment (LeoColomb) [reply docker-entrypoint.sh:13]: Arf, that looks a bit too much... the following might be sufficient right?
```suggestion
		printf 'Both %s and %s are set (but are exclusive)' "$var" "$fileVar"
```

File: docker-entrypoint.sh (hunk 2 above)
Comment (LeoColomb) [reply docker-entrypoint.sh:30]: Nope, it has to be documented but by default the docker script generate a random, which the best option.

Comment (jhollowe) [reply docker-entrypoint.sh:30]:
```suggestion
```

**Pull request merged**